ide/ui/ide-workspace/src/main/resources/META-INF/dirigible/ide-workspace/workspace.js

```diff
@@ -1097,46 +1107,104 @@ angular.module('workspace', ['workspace.config', 'ideUiCore', 'ngAnimate', 'ngSa
                                 }.bind(self, this)
                             }
                         };
+                        ctxmenu.create.submenu = new_submenu;
                     }
 
                     if (ctxmenu.create) {
+                        let nodeChildren = node.original._file.files.map(x => x.name);
+
                         for (let i = 0; i < priorityFileTemplates.length; i++) {
                             let fileTemplate = priorityFileTemplates[i];
+                            let isDisabled = false;
+                            if (fileTemplate.oncePerFolder) {
+                                for (let nc = 0; nc < nodeChildren.length; nc++) {
+                                    if (fileTemplate.nameless) {
+                                        if (nodeChildren[nc] === `.${fileTemplate.extension}`) {
+                                            isDisabled = true;
+                                            break;
+                                        }
+                                    } else if (nodeChildren[nc].endsWith(`.${fileTemplate.extension}`)) {
+                                        isDisabled = true;
+                                        break;
+                                    }
+                                }
+                            }
                             ctxmenu.create.submenu[fileTemplate.name] = {
+                                "_disabled": isDisabled,
                                 "separator_after": (i + 1 === priorityFileTemplates.length),
                                 "label": fileTemplate.label,
                                 "action": function (wnd, data) {
                                     let tree = $.jstree.reference(data.reference);
                                     let parentNode = tree.get_node(data.reference);
                                     let fileNode = {
-                                        type: 'file'
+                                        type: 'file',
+                                        data: fileTemplate.data,
                                     };
-                                    fileNode.text = 'file.' + fileTemplate.extension;
-                                    fileNode.data = fileTemplate.data;
-                                    tree.create_node(parentNode, fileNode, "last", function (new_node) {
-                                        tree.edit(new_node);
-                                    });
+                                    if (fileTemplate.nameless) fileNode.text = `.${fileTemplate.extension}`;
+                                    else fileNode.text = `file.${fileTemplate.extension}`;
```

Comment:
Please do `if-else-if-else` statements (even single lined), in the following way:
```javascript
if (fileTemplate.nameless) {
  fileNode.text = `.${fileTemplate.extension}`;
} else {
  fileNode.text = `file.${fileTemplate.extension}`;
}
```